diff --git a/vendor/magento/module-catalog/Controller/Category/View.php b/vendor/magento/module-catalog/Controller/Category/View.php
index a72f764..756c06a 100644
--- a/vendor/magento/module-catalog/Controller/Category/View.php
+++ b/vendor/magento/module-catalog/Controller/Category/View.php
@@ -152,7 +152,7 @@ class View extends \Magento\Framework\App\Action\Action
      */
     public function execute()
     {
-        if ($this->_request->getParam(\Magento\Framework\App\ActionInterface::PARAM_NAME_URL_ENCODED)) {
+        if ($this->_request->getParam(\Magento\Framework\App\ActionInterface::PARAM_NAME_URL_ENCODED) && !$this->_request->getParam('___from_store')) {
             return $this->resultRedirectFactory->create()->setUrl($this->_redirect->getRedirectUrl());
         }
         $category = $this->_initCategory();
diff --git a/vendor/magento/module-catalog/Controller/Product/View.php b/vendor/magento/module-catalog/Controller/Product/View.php
index 4c577eb..5149079 100644
--- a/vendor/magento/module-catalog/Controller/Product/View.php
+++ b/vendor/magento/module-catalog/Controller/Product/View.php
@@ -76,7 +76,7 @@ class View extends \Magento\Catalog\Controller\Product
         $productId = (int) $this->getRequest()->getParam('id');
         $specifyOptions = $this->getRequest()->getParam('options');
 
-        if ($this->getRequest()->isPost() && $this->getRequest()->getParam(self::PARAM_NAME_URL_ENCODED)) {
+        if ($this->getRequest()->isPost() && $this->getRequest()->getParam(self::PARAM_NAME_URL_ENCODED) && !$this->getRequest()->getParam('___from_store')) {
             $product = $this->_initProduct();
             if (!$product) {
                 return $this->noProductRedirect();
